codec-http/src/main/java/io/netty/handler/codec/http/ServerCookieDecoder.java

```diff
@@ -157,20 +119,52 @@
                     if (i == headerLen) {
                         // NAME (no value till the end of string)
                         newNameEnd = headerLen;
-                        value = null;
+                        newValueStart = newValueEnd = -1;
                         break;
                     }
                 }
             }
 
-            if (!rfc2965Style || (!header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) &&
-                    !header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) &&
-                    !header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
+            if (newValueStart == -1) {
+                logger.debug("Skipping cookie with null value");
+                continue;
+            }
+
+            if (rfc2965Style && (header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) ||
+                    header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) ||
+                    header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
 
                 // skip obsolete RFC2965 fields
-                String name = header.substring(newNameStart, newNameEnd);
-                cookies.add(new DefaultCookie(name, value));
+                continue;
+            }
+
+            String name = header.substring(newNameStart, newNameEnd);
+
+            Character c = firstInvalidCookieNameOctet(name);
+            if (c != null) {
+                logger.debug("Skipping cookie because name '{}' contains invalid char '{}'", name, c.charValue());
```

Comment:
Where do you see auto-boxing involved? Do you mean, once I'll return an int instead of a Character, as i'll have to cast to char?